### PR TITLE
optimize memory

### PR DESCRIPTION
**Pull request by SantyWang**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by fish9167** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot  run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

File: native/cocos/2d/renderer/RenderEntity.cpp [hunk 1]
```diff
@@ -28,41 +28,44 @@
 #include "bindings/utils/BindingUtils.h"
 
 namespace cc {
-RenderEntity::RenderEntity() {
+RenderEntity::RenderEntity(RenderEntityType type) : _renderEntityType(type) {
+    if (type == RenderEntityType::STATIC) {
+        ccnew_placement(&_staticDrawInfos) std::array<RenderDrawInfo, RenderEntity::STATIC_DRAW_INFO_CAPACITY>();
+    } else {
+        ccnew_placement (&_dynamicDrawInfos) ccstd::vector<RenderDrawInfo*>();
+    }
```
Comment (minggo) [inline native/cocos/2d/renderer/RenderEntity.cpp:36]: Why need do new here?

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Pull request merged**